# Fisher's Exact Test

Fisher's Exact Test is a statistical method used to determine if there are nonrandom associations between two categorical variables. This test is particularly valuable in scenarios involving small sample sizes and 2x2 contingency tables, where other tests like the Chi-Squared Test might not be reliable.

#### Business Scenario: Email Marketing Campaign Effectiveness

#### Data Set Description

A small business aims to compare the effectiveness of two different email marketing campaigns (A and B) concerning their response rates. Given budget constraints, only a limited number of emails can be sent, resulting in a small dataset. The outcomes are categorized into "Response" and "No Response."

The data is summarized in a 2x2 contingency table:

|             | Response | No Response |
|-------------|----------|-------------|
| **Campaign A** | x        | y           |
| **Campaign B** | z        | w           |

Here, x, y, z, and w are the counts of outcomes for each combination of campaign type and response.

#### Business Problem

The business needs to identify which email campaign is more effective, specifically determining if the difference in response rates between Campaign A and Campaign B is statistically significant, especially considering the small sample size.

#### Applying Fisher's Exact Test

Fisher's Exact Test is the ideal choice for this scenario, given the binary nature of both the independent variable (Campaign Type) and the dependent variable (Response). The test assesses the exact likelihood of observing the given distribution of responses, under the null hypothesis that both campaigns are equally effective.

#### Steps:

1. **Construct the 2x2 Contingency Table:** Organize observed counts of responses and no responses for Campaigns A and B.
2. **Apply Fisher's Exact Test:** Calculate the probability of observing the given distribution of responses, assuming no difference in campaign effectiveness.
3. **Interpret the Results:** A p-value less than 0.05 typically indicates a statistically significant difference in campaign effectiveness, suggesting one campaign elicited a higher response rate than the other.


In [3]:
from scipy.stats import fisher_exact
import numpy as np

# Simulated data for the 2x2 contingency table
contingency_table = np.array([
    [30, 70],  # Campaign A: 30 responses, 70 no responses
    [45, 55]   # Campaign B: 45 responses, 55 no responses
])

# Applying Fisher's Exact Test
odds_ratio, p_value = fisher_exact(contingency_table)

print(f'Odds Ratio: {odds_ratio:.4f}, P-Value: {p_value:.4}')

Odds Ratio: 0.5238, P-Value: 0.04053


#### Interpretation of Fisher's Exact Test Results

The Fisher's Exact Test applied to the email marketing campaign effectiveness scenario yielded the following results:

- **Odds Ratio**: 0.524  
- **P-Value**: 0.0405  

Given the p-value is approximately 0.0405, which is less than the conventional significance level of 0.05, we conclude that there is a statistically significant difference in the effectiveness of the two email campaigns. This suggests that Campaign B led to a higher response rate than Campaign A, given the odds ratio is less than 1, indicating Campaign A's response odds are lower compared to Campaign B.

Outcome and Strategic Actions:
The statistical significance indicates that the small business should consider prioritizing or further analyzing the strategies employed in Campaign B, as it appears to be more effective in eliciting responses from the target audience. This finding can guide future marketing strategies, potentially leading to adjustments in how email campaigns are designed and targeted to maximize engagement and response rates.